core/controllers/admin.py

```diff
@@ -2643,3 +2643,77 @@ def get(self) -> None:
         ]
 
         self.render_json({'interaction_ids': interaction_ids})
+
+
+class QuestionAdminRoleHandlerNormalizedPayloadDict(TypedDict):
+    """Dict representation of QuestionAdmin's normalized_payload
+    dictionary.
+    """
+
+    username: str
+    action: str
+
+
+class QuestionAdminRoleHandler(
+    base.BaseHandler[
+        QuestionAdminRoleHandlerNormalizedPayloadDict, Dict[str, str]
+    ]
+):
+    """Handler to assign or deassigning a question admin."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'PUT': {
+            'username': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'action': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': ['assign', 'deassign']
+                }
+            }
+        }
+    }
+
+    @acl_decorators.can_access_admin_page
+    def put(self) -> None:
+        """Adds or removes the question role for a user.
+
+        Raises:
+            NotFoundException. User with given username does not exist.
+        """
+        assert self.normalized_payload is not None
+        username = self.normalized_payload['username']
+        action = self.normalized_payload['action']
+
+        user_settings = user_services.get_user_settings_from_username(username)
+
+        if user_settings is None:
+            raise self.NotFoundException(
+                'User with given username does not exist.')
+
+        user_id = user_settings.user_id
+        if action == 'assign':
+            if not feconf.ROLE_ID_QUESTION_ADMIN in user_settings.roles:
```

Comment:
Try to reduce nesting if you can.

For example, if we have some code like this:
```python
if condition_A:
    if condition_B:
        # do something
```
we can reduce the nesting by refactoring the above code to this:
```python
if condition_A and condition_B:
    # do something
```
which is easier to read.